echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/QuietPeriodIndicator.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.pipelinetriggers;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.patterns.PolledMeter;
+import com.netflix.spinnaker.echo.config.QuietPeriodIndicatorConfigurationProperties;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+import static java.time.format.DateTimeFormatter.ISO_INSTANT;
+
+@Slf4j
+@Component
+@ConditionalOnExpression("${quietPeriod.enabled:false}")
+@Scope(value = ConfigurableBeanFactory.SCOPE_SINGLETON)
+public class QuietPeriodIndicator {
+  private final Registry registry;
+
+  private final boolean enabled;
+  private final long startEpochMillis;
+  private final long endEpochMillis;
+  private final List<String> suppressedTriggerTypes;
+
+  private final Id quietPeriodTestId;
+
+  @Autowired
+  public QuietPeriodIndicator(@NonNull Registry registry,
+                              @NonNull QuietPeriodIndicatorConfigurationProperties config) {
+    this.registry = registry;
+
+    long startEpochMillis = parseIso(config.getStartIso());
+    long endEpochMillis = parseIso(config.getEndIso());
+    this.startEpochMillis = startEpochMillis;
+    this.endEpochMillis = endEpochMillis;
+    this.suppressedTriggerTypes = config.getSuppressedTriggerTypes();
+
+    this.enabled = config.isEnabled() && (startEpochMillis > 0 && endEpochMillis > 0);
+    if (this.enabled) {
+      log.info("Enabling quiet periods.  Span in millis: {} to {}, ISO {} to {} (inclusive)",
+               startEpochMillis, endEpochMillis, config.getStartIso(), config.getEndIso());
+      log.info("Will suppress triggers of types {} during quiet periods.", config.getSuppressedTriggerTypes());
+    } else {
+      log.info("Not enabling quiet periods.");
```

Comment:
And probably drop this else branch entirely.

Reply:
done.